### PR TITLE
Record arguments used when method not found exception is thrown

### PR DESCRIPTION
**Pull request by ciaranmcnulty**
This will be useful when handling these exceptions in PhpSpec (e.g. by generating a method)

### CODE REVIEW

File: spec/Prophecy/Exception/Doubler/MethodNotFoundExceptionSpec.php [hunk 1]
```diff
@@ -9,7 +9,7 @@ class MethodNotFoundExceptionSpec extends ObjectBehavior
 {
     function let()
     {
-        $this->beConstructedWith('', 'User', 'getName');
+        $this->beConstructedWith('', 'User', 'getName', array(1,2,3));
```
Comment (GrahamCampbell) [inline spec/Prophecy/Exception/Doubler/MethodNotFoundExceptionSpec.php]: Is the cs intentionally incorrect here? `array(1, 2, 3)` is what we want isn't it?

File: src/Prophecy/Exception/Doubler/MethodNotFoundException.php [hunk 2]
```diff
@@ -15,18 +15,24 @@ class MethodNotFoundException extends DoubleException
 {
     private $classname;
     private $methodName;
+    /**
+     * @var array|null
```
Comment (stof) [inline src/Prophecy/Exception/Doubler/MethodNotFoundException.php]: this should always be an array, not null IMO

File: src/Prophecy/Exception/Doubler/MethodNotFoundException.php [hunk 3]
```diff
@@ -15,18 +15,24 @@ class MethodNotFoundException extends DoubleException
...
+    private $arguments;
 
     /**
      * @param string $message
      * @param string $classname
      * @param string $methodName
+     * @param array  $arguments
      */
-    public function __construct($message, $classname, $methodName)
+    public function __construct($message, $classname, $methodName, $arguments = array())
```
Comment (stof) [inline src/Prophecy/Exception/Doubler/MethodNotFoundException.php]: please typehint the new argument as array

File: src/Prophecy/Prophecy/MethodProphecy.php [hunk 4]
```diff
@@ -46,9 +46,10 @@ public function __construct(ObjectProphecy $objectProphecy, $methodName, $argume
     {
         $double = $objectProphecy->reveal();
         if (!method_exists($double, $methodName)) {
+            var_dump(2,$arguments);
```
Comment (stof) [inline src/Prophecy/Prophecy/MethodProphecy.php]: this must be removed

Comment (ciaranmcnulty) [inline src/Prophecy/Prophecy/MethodProphecy.php]: eeeesh, what an amateur

**Comment by everzet** [comment]
Oh LOL

**Pull request merged**